test/test_counterfactual.py

```diff
@@ -0,0 +1,76 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+import pytest
+
+import numpyro
+import numpyro.distributions as dist
+import numpyro.handlers as handlers
+from numpy.testing import assert_allclose
+
+
+@pytest.mark.parametrize('intervene,observe,flip', [
+    (True, False, False),
+    (False, True, False),
+    (True, True, False),
+    (True, True, True),
+])
+def test_counterfactual_query(intervene, observe, flip):
```

Comment:
Thanks for adding this test!

nit: I think we can move this to `test_handlers` file.